raytracing/__main__.py

```diff
@@ -1,430 +1,123 @@
+        self.lexer = pygments.lexers.get_lexer_by_name('python', stripall=True)
+
+        self.parseArguments()
+
+    def exampleCarousel(self, exampleIndexList):
+        for i in exampleIndexList:
+            self.selectedFileIndex = i
+            self.selectedFile = self.exampleDirPath + '\example{}.py'.format(i)
+            self.generateExample()
+            self.showExample()
+
+    def showExample(self):
+        webbrowser.open("file://" + self.htmlTemporaryExampleFile)
+
+    def generateExample(self):
+        self.createTemporaryHTML()
+        self.generateFigureFromCode()
+        self.generateHTMLHighlightedCode()
+        self.generateHTMLDescription()
+        self.mergeComponentsToHTML()
+
+    def mergeComponentsToHTML(self):
+        with open(self.htmlTemporaryExampleFile, 'a+') as f:
+            f.write(self.HTMLDescription)
+            f.write(self.HTMLFigure)
+            f.write(self.HTMLHighlightedCode)
+            f.close()
+
+    def createTemporaryHTML(self):
+        temp = tempfile.NamedTemporaryFile(delete=False)
+        path = temp.name + '.html'
+        with open(path, 'w') as f:
+            f.write(''''<head><link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous"></head>''')
+
+        self.htmlTemporaryExampleFile = path
+
+    def generateFigureFromCode(self):
+        self.runExampleCode()
+        fig, axes = plt.subplots(figsize=(7, 5))
+        self.figureObject.createRayTracePlot(axes=axes)
+        with tempfile.TemporaryFile(suffix=".png") as tmpfile:
+            plt.savefig(tmpfile, format="png")
+            tmpfile.seek(0)
+            self.base64Figure = base64.b64encode(tmpfile.read()).decode()
+        self.HTMLFigure = ''''<div style="display:inline-block; vertical-align:top"><img class="icon" src="data:image/png;base64,{} "></div>'''.format(
+            self.base64Figure)
+
+    def generateHTMLHighlightedCode(self, code=""):
+        if not code:
+            code = self.getExampleCode()
+            code = re.sub("'''\n*?(.+?[\s\S]+?)\n*?'''\n*", "", code)
+        self.HTMLHighlightedCode = highlight(code, self.lexer, self.formatter)
+
+        return self.HTMLHighlightedCode
+
+    def generateHTMLDescription(self):
+        code = self.getExampleCode()
+        docString = self.getDocstring(code)
+        self.HTMLDescription = '''<div class="alert alert-success" role="alert"><h4 class="alert-heading">Example Description</h4><div style="word-wrap: break-word;
+   width: 600px;">{}</div></div>'''.format(docString)
+
+    def runExampleCode(self):
+        with patch('matplotlib.pyplot.show') as p:
+            global path
+            exec(open(self.selectedFile).read(), globals())
+            self.figureObject = path
```

Comment:
I don't think we should call this ImagingPath a figureObject since it's not.

Reply:
very true.